internal/task/task/monitor/create_container.go

```diff
@@ -0,0 +1,173 @@
+			"web.console.templates":       "/usr/share/prometheus/consoles",
+			"web.listen-address":          fmt.Sprintf(":%d", cfg.GetListenPort()),
+		}
+	}
+	args := []string{}
+	for k, v := range argsMap {
+		var item string
+		if v != nil {
+			item = fmt.Sprintf("--%s=%v", k, v)
+		} else {
+			item = fmt.Sprintf("--%s", k)
+		}
+		args = append(args, item)
+	}
+	return strings.Join(args, " ")
+}
+
+func getMountVolumes(cfg *configure.MonitorConfig) []step.Volume {
+	role := cfg.GetRole()
+	volumes := []step.Volume{}
+	switch role {
+	case ROLE_NODE_EXPORTER:
+		volumes = append(volumes, step.Volume{
+			HostPath:      "/",
+			ContainerPath: "/host:ro,rslave",
+		},
+			step.Volume{
+				HostPath:      "/run/udev/data",
+				ContainerPath: "/run/udev/data",
+			},
+			step.Volume{
+				HostPath:      "/run/dbus/system_bus_socket",
+				ContainerPath: "/var/run/dbus/system_bus_socket:ro",
+			})
+	case ROLE_PROMETHEUS:
+		volumes = append(volumes, step.Volume{
+			HostPath:      cfg.GetDataDir(),
+			ContainerPath: "/prometheus",
+		})
+	case ROLE_GRAFANA:
+		volumes = append(volumes, step.Volume{
+			HostPath:      cfg.GetDataDir(),
+			ContainerPath: "/var/lib/grafana",
+		})
+	}
+	return volumes
+}
+
+func getEnvironments(cfg *configure.MonitorConfig) []string {
+	role := cfg.GetRole()
+	if role == ROLE_GRAFANA {
+		return []string{
+			"GF_INSTALL_PLUGINS=grafana-piechart-panel",
+			fmt.Sprintf("GF_SECURITY_ADMIN_USER=%s", cfg.GetGrafanaUser()),
+			fmt.Sprintf("GF_SECURITY_ADMIN_PASSWORD=%s", cfg.GetGrafanaPassword()),
+			fmt.Sprintf("GF_SERVER_HTTP_PORT=%d", cfg.GetListenPort()),
+		}
+	}
+	return []string{}
+}
+
+func NewCreateContainerTask(curveadm *cli.CurveAdm, cfg *configure.MonitorConfig) (*task.Task, error) {
+	host := cfg.GetHost()
+	hc, err := curveadm.GetHost(host)
+	if err != nil {
+		return nil, err
+	}
+
+	// new task
+	subname := fmt.Sprintf("host=%s role=%s", host, cfg.GetRole())
+	t := task.NewTask("Create Container", subname, hc.GetSSHConfig())
+
+	// add step to task
+	var oldContainerId, containerId string
+	clusterId := curveadm.ClusterId()
+	mcId := cfg.GetId()
+	serviceId := curveadm.GetServiceId(mcId)
+	kind := cfg.GetKind()
+	role := cfg.GetRole()
+	hostname := fmt.Sprintf("%s-%s-%s", kind, role, serviceId)
+	options := curveadm.ExecOptions()
+	options.ExecWithSudo = false
+
+	t.AddStep(&common.Step2GetService{ // if service exist, break task
+		ServiceId:   serviceId,
+		ContainerId: &oldContainerId,
+		Storage:     curveadm.Storage(),
+	})
+	t.AddStep(&step.CreateDirectory{
+		Paths:       []string{cfg.GetDataDir()},
+		ExecOptions: options,
+	})
+	t.AddStep(&step.CreateContainer{
+		Image:       cfg.GetImage(),
+		Command:     getArguments(cfg),
+		AddHost:     []string{fmt.Sprintf("%s:127.0.0.1", hostname)},
+		Envs:        getEnvironments(cfg),
+		Hostname:    hostname,
+		Init:        true,
+		Name:        hostname,
+		Privileged:  true,
+		User:        "0:0",
+		Pid:         "host",
```

Comment:
Pid using `host` will share host progress.

Reply:
Yes, now the container created by curveadm are all 'host' type.